### PR TITLE
Acacia keyboard

### PR DESCRIPTION
**Pull request by jpsecher**
## Description

New custom 42 key keyboard, with link to all source files to build it.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/acacia/info.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+    "keyboard_name": "Acacia",
+    "keyboard_folder": "handwired/acacia",
```
Comment (zvecr) [inline keyboards/handwired/acacia/info.json:3]:
```suggestion
```

File: keyboards/handwired/acacia/readme.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    make handwired/acacia
+
+or
+
+    qmk compile -kb handwired/acacia -km default
+
+To clean up:
+
+    qmk clean
```
Comment (zvecr) [inline keyboards/handwired/acacia/readme.md:18]:
```suggestion
Make example for this keyboard (after setting up your build environment):

    make handwired/acacia:default

Flashing example for this keyboard:

    make handwired/acacia:default:flash
```

File: keyboards/handwired/acacia/rules.mk [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+VIA_ENABLE = no             # GUI configuration
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no            # N-Key Rollover
+BACKLIGHT_ENABLE = no
+RGBLIGHT_ENABLE = no
+RGB_MATRIX_ENABLE = no
+AUDIO_ENABLE = no
+SPLIT_KEYBOARD = no
+OLED_ENABLE = no
+LTO_ENABLE = yes            # Link-time optimisation for smaller code
```
Comment (zvecr) [inline keyboards/handwired/acacia/rules.mk:14]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # N-Key Rollover
LTO_ENABLE = yes            # Link-time optimisation for smaller code
```

File: keyboards/handwired/acacia/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+//#define BS_CTL LCTL_T(KC_BSPC)
+#define BS_GUI LGUI_T(KC_BSPC)
+//#define BS_NAV LT(_NAVI_FN, KC_BSPC)
+#define BS_SYM LT(_SYM_NUM, KC_BSPC)
+//#define DEL_SYM LT(_SYM_NUM, KC_DEL)
+//#define DEL_NAV LT(_NAVI_FN, KC_DEL)
+#define SPC_NAV LT(_NAVI_FN, KC_SPC)
+#define OSM_AGR OSM(MOD_RALT)
+//#define OSM_MEH OSM(MOD_MEH)
+//#define OSM_CG OSM(MOD_LCTL|MOD_LGUI)
```
Comment (zvecr) [inline keyboards/handwired/acacia/keymaps/default/keymap.c:38]: We wouldnt really want commented out code here

File: keyboards/handwired/acacia/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+enum layer_names {
+    _QWERTY,
+    _NAVI_FN,
+    _SYM_NUM,
+    _MACRO,
+    NUMBER_OF_LAYERS
```
Comment (zvecr) [inline keyboards/handwired/acacia/keymaps/default/keymap.c:45]:
```suggestion
    _MACRO
```

File: keyboards/handwired/acacia/acacia.c [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
```
Comment (zvecr) [inline keyboards/handwired/acacia/acacia.c:15]:
```suggestion
 */

#include "acacia.h"
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**